src/Client/ConcordiumClient.cs

```diff
@@ -205,6 +205,52 @@ public ITransactionStatus GetBlockItemStatus(TransactionHash transactionHash, Ca
         return TransactionStatusFactory.CreateTransactionStatus(blockItemStatus);
     }
 
+    /// <summary>
+    /// Wait until the transaction is finalized.
+    /// </summary>
+    /// <param name="transactionHash">Transaction Hash which is included in blocks returned.</param>
+    /// <param name="token">Cancellation token</param>
+    /// <returns>Hash of the block finalizing the transaction and the block summary of the transaction.</returns>
+    /// <exception cref="RpcException">RPC error occurred, access <see cref="RpcException.StatusCode"/> for more information.</exception>
+    public async Task<(BlockHash BlockHash, BlockItemSummary Summary)> WaitUntilFinalized(
+        TransactionHash transactionHash,
+        CancellationToken token = default
+    )
+    {
+        static bool isFinalized(ITransactionStatus status, out (BlockHash, BlockItemSummary)? output)
+        {
+            switch (status)
+            {
+                case TransactionStatusFinalized finalized:
+                    output = finalized.State;
+                    return true;
+                default:
+                    output = null;
+                    return false;
+            };
+        }
+
+        // Check the transaction status straight away and return if finalized already.
+        var status = await this.GetBlockItemStatusAsync(transactionHash, token);
```

Comment:
If the transaction is missing, does this raise an exception or return "not found"?

Reply:
It will be an RpcException with status of not found

Reply:
Ok, there is a bit of a race condition in common usage "send tx + wait for finalization" so it'd be useful to document this behavior that if the transaction is not yet there when this is called it'll fail.